projects/laji/src/app/+project-form/about/dataset-about/dataset-about.component.ts

```diff
@@ -1,12 +1,42 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { WarehouseApi } from '../../../shared/api/WarehouseApi';
 import { Form } from '../../../shared/model/Form';
+import { WarehouseQueryInterface } from '../../../shared/model/WarehouseQueryInterface';
+
+interface DatasetStats {
+  count?: number;
+  speciesCount?: number;
+  firstLoadDateMax?: string;
+}
 
 @Component({
   selector: 'laji-dataset-about',
   templateUrl: './dataset-about.component.html',
   styleUrls: ['./dataset-about.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DatasetAboutComponent {
+export class DatasetAboutComponent implements OnChanges {
   @Input() form: Form.SchemaForm;
+
+  stats$?: Observable<DatasetStats>;
+
+  constructor(
+    private warehouseService: WarehouseApi
+  ) {}
+
+  ngOnChanges() {
+    if (this.form?.collectionID) {
+      const query: WarehouseQueryInterface = {
+        collectionId: [this.form.collectionID],
+        taxonCounts: true
+      };
+      this.stats$ = this.warehouseService.warehouseQueryAggregateGet(query, [], [], 1, 1, false, false).pipe(
+        map(res => res.results[0] || {})
+      );
+    } else {
+      this.stats$ = null;
+    }
+  }
```

Comment:
To simplify this, I think it's a fair assumption that `this.form` is defined. If it's not and an error occurs, that lets us know that there is a dataset-about component used without a form which implies a larger problem in the overall design. That way we can drop the if-else and don't have to set `stats$` to `null`, which shouldn't be done in reactive programming (RJXS). Proper way would be do make `stats$` emit an empty value.

I didn't know this beforehand, but after some reading it seems that `@Input`s can be marked required in the `@Component` `selector` like so:

```
@Component({
  selector: 'laji-dataset-about[form]',
  ...
```

We might want to add that, and start using that syntax more systemitically.

Reply:
Hmm I didn't take into account that the `if (this.form?.collectionID) {` is checking also for the collectionID and not only the `form`... In practice in our problem domain here we don't necessarily have to care about that since the dataset forms always have a collectionID, but it's not a bad idea to care about it.

In that case the `stats$` shouldn't be set to `null`. Also, if we are thinking out of our problem domain and thinking in theoretical terms, the `ngOnChange` could run multiple times and in reactive programming we don't want to reinitialize the whole pipe every time, but make the same singular pipe emit the next value. So I think the cleanest solution would be wrap collectionID to an observable, and make `stats$` react to that.

Reply:
The logic is now changed such that `stats$` is initialized only once. I didn't know either that it's possible to mark `@Input` required with the selector, seems useful.